src/main/java/org/entando/kubernetes/model/bundle/processor/PluginProcessor.java

```diff
@@ -150,10 +150,40 @@ private void logDescriptorWarnings(PluginDescriptor descriptor) {
             log.warn(errMessage,
                     descriptor.getDockerImage(),
                     BundleUtilities.MAX_ENTANDO_K8S_POD_NAME_LENGTH,
-                    BundleUtilities.truncatePodPrefixName(deploymentBaseName));
+                    this.truncatePodPrefixName(deploymentBaseName));
         }
     }
 
+    public String composeFullDeploymentNameAndTruncateIfNeeded(PluginDescriptor descriptor, String bundleId) {
```

Comment:
@firegloves simplify the naming, `generateFullDeploymentName` already implies all the operations required to make a valid full deployment name.
If you wish to add details, add a javadoc